apps/ledger-live-desktop/src/renderer/modals/EditTransaction/Body.tsx

```diff
@@ -0,0 +1,290 @@
+
+  const {
+    transaction,
+    setTransaction,
+    updateTransaction,
+    account,
+    parentAccount,
+    status,
+    bridgeError,
+    bridgePending,
+  } = useBridgeTransaction(() => {
+    const parentAccount = params && params.parentAccount;
+    const account = (params && params.account) || accounts[0];
+    return {
+      account,
+      parentAccount,
+      transaction: fromTransactionRaw(params.transactionRaw as EthereumTransactionRaw),
+    };
+  });
+  const [optimisticOperation, setOptimisticOperation] = useState<Operation | null>(null);
+  const [transactionError, setTransactionError] = useState<Error | null>(null);
+  const [signed, setSigned] = useState(false);
+
+  const currency = account ? getAccountCurrency(account) : undefined;
+
+  const currencyName = currency ? currency.name : undefined;
+
+  const handleCloseModal = useCallback(() => {
+    closeModal("MODAL_EDIT_TRANSACTION");
+  }, [closeModal]);
+
+  const handleRetry = useCallback(() => {
+    setTransactionError(null);
+    setOptimisticOperation(null);
+    setSigned(false);
+  }, []);
+
+  const handleTransactionError = useCallback((error: Error) => {
+    if (!(error instanceof UserRefusedOnDevice)) {
+      logger.critical(error);
+    }
+    setTransactionError(error);
+  }, []);
+
+  const handleOperationBroadcasted = useCallback(
+    (optimisticOperation: Operation) => {
+      if (!account) return;
+      const mainAccount = getMainAccount(account, parentAccount);
+      updateAccountWithUpdater(mainAccount.id, account =>
+        addPendingOperation(account, optimisticOperation),
+      );
+      setOptimisticOperation(optimisticOperation);
+      setTransactionError(null);
+    },
+    [account, parentAccount, updateAccountWithUpdater],
+  );
+
+  const handleStepChange = useCallback(e => onChangeStepId(e.id), [onChangeStepId]);
+  const error = transactionError || bridgeError;
+  if (!account) return null;
+  const mainAccount = getMainAccount(account, parentAccount);
+  const feePerGas = new BigNumber(
+    EIP1559ShouldBeUsed(mainAccount.currency)
+      ? (params.transactionRaw as EthereumTransactionRaw).maxFeePerGas ?? "0"
+      : (params.transactionRaw as EthereumTransactionRaw).gasPrice ?? "0",
+  );
+  const feeValue = new BigNumber(
+    (params.transactionRaw as EthereumTransactionRaw).userGasLimit ||
+      (params.transactionRaw as EthereumTransactionRaw).estimatedGasLimit ||
+      0,
+  )
+    .times(feePerGas)
+    .div(new BigNumber(10).pow(mainAccount.unit.magnitude));
+  const haveFundToCancel = mainAccount.balance.gt(feeValue.times(1.3));
+  const haveFundToSpeedup = mainAccount.balance.gt(
+    feeValue.times(1.1).plus(account.type === "Account" ? params.transactionRaw.amount : 0),
+  );
+  // log account and fees info
+  logger.log(`main account address: ${mainAccount.freshAddress}`);
+  logger.log(`main account balance: ${mainAccount.balance.toNumber()}`);
+  logger.log(`feeValue: ${feeValue.toNumber()}`);
```

Comment:
use `.toFixed()` instead in case it's above the Number.MAX_SAFE_INTEGER limit.

Reply:
fixed